addons/pos_self_order/static/src/self_order_service.js

```diff
@@ -67,7 +66,14 @@ export class SelfOrder {
             );
         }
 
-        this.productsGroupedByTag = groupBy(this.products, "tag");
+        this.productsGroupedByCategory = this.products.reduce((acc, product) => {
+            if (!acc[product.pos_categ_id.name]) {
+                acc[product.pos_categ_id.name] = [];
+            }
+
+            acc[product.pos_categ_id.name].push(product);
+            return acc;
+        }, {});
```

Comment:
```suggestion
        this.productsGroupedByCategory = groupBy(this.products, (product) => product.pos_categ_id.name)
```
I think you can keep the use of `groupBy`.